timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -117,7 +118,16 @@ private void scheduleExpiredLockReaper() {
     }
 
     public AsyncResult<Leased<LockToken>> lock(UUID requestId, Set<LockDescriptor> lockDescriptors, TimeLimit timeout) {
-        return heldLocks.getExistingOrAcquire(requestId, () -> acquireLocks(requestId, lockDescriptors, timeout));
+        return lock(requestId, lockDescriptors, timeout, Optional.empty());
+    }
```

Comment:
Commenting once to clarify: For any public/interface method that passes metadata down the call stack, we add an overload that takes an `Optional<LockRequestMetadata>`. We copy over the original method body, but change invocation of downstream methods to the version including metadata. Then, we make the original method (the one without metadata) just delegate to the new method.